File: Android/app/src/main/java/com/khandoba/securedocs/config/AppConfig.kt
package com.khandoba.securedocs.config

object AppConfig {
    // App Information
    const val APP_VERSION = "1.0.1"
    const val APP_BUILD_NUMBER = 30
    const val APP_NAME = "Khandoba Secure Docs"
    
    // Backend Configuration
    // Using same Supabase database as iOS app for cross-platform sync
    const val USE_SUPABASE = true // Set to true to use Supabase (same DB as iOS)
    const val SUPABASE_URL = "https://uremtyiorzlapwthjsko.supabase.co"
    const val SUPABASE_ANON_KEY = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6InVyZW10eWlvcnpsYXB3dGhqc2tvIiwicm9sZSI6ImFub24iLCJpYXQiOjE3NjU5NjI3MDcsImV4cCI6MjA4MTUzODcwN30.P4Yg4Gl040Msv0TeRuHQ-_SuGfeNEHCV234W5TTSN7Y"
    
    // Supabase Storage Buckets (same as iOS)
    const val ENCRYPTED_DOCUMENTS_BUCKET = "encrypted-documents"
    const val VOICE_MEMOS_BUCKET = "voice-memos"
    const val INTEL_REPORTS_BUCKET = "intel-reports"
    
    // Real-time Configuration
    const val ENABLE_REALTIME = true
    
    // Firebase Configuration (if not using Supabase)
    const val FIREBASE_PROJECT_ID = "khandoba-secure-docs"
    
    // Feature Flags
    const val ENABLE_ANALYTICS = true
    const val ENABLE_CRASH_REPORTING = true
    const val ENABLE_PUSH_NOTIFICATIONS = true
    
    // Security
    const val REQUIRE_BIOMETRIC_AUTH = true
    const val SESSION_TIMEOUT_MINUTES = 30
    const val MAX_LOGIN_ATTEMPTS = 5
    
    // App Group/Shared Preferences
    const val SHARED_PREFS_NAME = "khandoba_secure_docs_prefs"
    const val PERMISSIONS_SETUP_COMPLETE_KEY = "permissions_setup_complete"
}

File: platforms/apple/Khandoba Secure Docs/Config/AppConfig.swift
//
//  AppConfig.swift
//  Khandoba Secure Docs
//
//  Created by Jai Deshmukh on 12/2/25.
//

import Foundation
import Combine

struct AppConfig {
    // PRODUCTION MODE - Real Apple Sign In required
    static let isDevelopmentMode = false
    
    // App Information
    static let appVersion = "1.0.1"
    static let appBuildNumber = "30"
    static let appName = "Khandoba Secure Docs"
    
    // Production Configuration
    // Note: Supabase is used for platform-agnostic vault access (web, iOS, Android, etc.)
    // CloudKit is available as fallback for iOS-only features
    static let cloudKitContainer = "iCloud.com.khandoba.securedocs"
    
    // CloudKit API Configuration (fallback for iOS-only features)
    static let cloudKitKeyID = "PR62QK662L"
    static let cloudKitTeamID = "Q5Y8754WU4"
    static let cloudKitKeyPath = "AuthKey_PR62QK662L.p8" // Relative to project root
    
    // Supabase Configuration
    // Production Supabase project configured for platform-agnostic access
    // Set to true to use Supabase (enables web, iOS, Android, and other platform access)
    // Set to false to use CloudKit/SwiftData (iOS-only, Apple's managed service)
    static let useSupabase = true // Feature flag to switch between CloudKit and Supabase
    static let supabaseURL = SupabaseConfig.currentEnvironment.supabaseURL
    static let supabaseAnonKey = SupabaseConfig.currentEnvironment.supabaseAnonKey
    
    // Feature Flags
    static let enableAnalytics = true
    static let enableCrashReporting = true
    static let enablePushNotifications = true
    
    // Security
    static let requireBiometricAuth = true
    static let sessionTimeoutMinutes = 30
    static let maxLoginAttempts = 5
    
    // Development user credentials (only used if isDevelopmentMode = true)
    static let devUserID = "dev-user-123"
    static let devUserName = "Developer User"
    static let devUserEmail = "dev@khandoba.local"
    
    // Admin role removed - autopilot mode (ML handles everything)
    
    // App Group Identifier (must match extension)
    static let appGroupIdentifier = "group.com.khandoba.securedocs"
}
